### DISCUSSION TITLE
How do I pre-enable an event listener provider in realms using `kc.sh`?

### DISCUSSION THREAD

**Discussion by atrauzzi** [keycloak/keycloak#35362] [Q&A] [answered]
My event listener successfully builds, installs and works in Keycloak.  But I currently have to manually go into the admin UI to enable it on new installations:

![image](https://github.com/user-attachments/assets/da85465a-2575-4944-a425-7f60fc3a9762)

Is there any way I can provide the list of pre-enabled listeners for my Keycloak installs?

I had originally thought something like this:

```
/opt/keycloak/bin/kc.sh build \
    --verbose \
    --spi-events-listener-[my-listener-name-here]-enabled=true
```

But that doesn't seem to be working...

_[Question on Keycloak Discourse.](https://keycloak.discourse.group/t/how-do-i-pre-enable-an-event-listener-provider-in-realms-using-kc-sh/28968)_

**Comment by stianst** [comment, marked as answer]
You can make your event listener a global event listener by overriding the `isGlobal` method

https://github.com/keycloak/keycloak/blob/7e19ae47a4a136268c8f3a5fd9cae03f705ee53e/server-spi-private/src/main/java/org/keycloak/events/EventListenerProviderFactory.java#L27-L29

**Reply by atrauzzi** [reply]
Thank you @stianst !  Is this available in `26.0.6`?

I'm looking at what I currently have and that method isn't there.

![image](https://github.com/user-attachments/assets/40fd3011-8ed0-4399-9c92-d7eb1a41be1e)

**Reply by stianst** [reply]
Ah, sorry. This will not be available until 26.1, which is due to be released early January

**Reply by atrauzzi** [reply]
I'll keep an eye out! Thank you!

**Reply by atrauzzi** [reply]
@stianst -- Sorry, one more question...  I noticed that with `isGlobal`, the event listener doesn't show up in the list of event listeners, though it does appear to be enabled.

I think this will work fine for what I wanted, but I was wondering if there was actually a way to do this in such a way that if a realm wanted, it could still have the option to disable?

Basically, I want my listener to be pre-registered in much the same way that the `jboss-logging` one is for new realms.
